stacktrace.CallerFrames returns unexpected frame when called with nil

## Description

The `stacktrace.CallerFrames` function unexpectedly returns one frame when called with a `nil` pointer. This violates the expected behavior that no frames should be returned when the input is `nil`.

## Steps to Reproduce

1. Run the test (see the attached test code below).

## Expected Behavior

When `stacktrace.CallerFrames(nil)` is called, the loop should not iterate, and the variable `n` should remain 0.

## Actual Behavior

The test fails because `n` is 1, indicating that one frame was unexpectedly returned.

## Test Code

```go
	t.Run("nil", func(t *testing.T) {
		n := 0
		for range stacktrace.CallerFrames(nil) {
			n++
		}
		if n != 0 {
			t.Errorf("n must be 0, but %d", n)
		}
	})
```

## Test Output

```
=== RUN  TestCallersFrames/nil
   stacktrace_go123_test.go:46: n must be 0, but 1
--- FAIL: TestCallersFrames (0.00s)
   --- FAIL: TestCallersFrames/nil (0.00s)
```


gemma3:27b

--- a/v2/stacktrace_go123.go
+++ b/v2/stacktrace_go123.go
@@ -15,6 +15,9 @@ import (
 // and provides each frame through the yield function.
 func CallersFrames(callers []uintptr) iter.Seq[*runtime.Frame] {
 	return func(yield func(*runtime.Frame) bool) {
+		if len(callers) == 0 {
+			return
+		}
 		frames := runtime.CallersFrames(callers)
 		for {
 			frame, more := frames.Next()

--- a/v2/stacktrace_go123_test.go
+++ b/v2/stacktrace_go123_test.go
@@ -36,6 +36,16 @@ func TestCallersFrames(t *testing.T) {
 	if !slices.Equal(list, want) {
 		t.Errorf("list must be equal to callers; list=%v callers=%v", list, callers)
 	}
+
+	t.Run("nil", func(t *testing.T) {
+		n := 0
+		for range stacktrace.CallersFrames(nil) {
+			n++
+		}
+		if n != 0 {
+			t.Errorf("n must be 0, but %d", n)
+		}
+	})
 }
 
 func funcnameOf(t *testing.T, pc uintptr) string {

